Bluepill-cli/bp_test_dumper/bp_test_dumper.m

```diff
@@ -0,0 +1,90 @@
+//
+//  bp_test_dumper.m
+//  bp_test_dumper
+//
+//  Created by Ravi K. Mandala on 10/23/19.
+//  Copyright © 2019 LinkedIn. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+
+NSString *getSimpleName(NSString *testName) {
+    // Extracting test name from a string like "-[FeedHighlightedUpdateTrackingTest testFeedHighlightedUpdateEvent]"
+    NSString *simpleTestName = [testName componentsSeparatedByString:@" "][1];
+    simpleTestName = [simpleTestName substringToIndex:simpleTestName.length-1];
+    return simpleTestName;
+}
+
+/* Return code:
+ * 0 - Successful
+ * 1 - Error loading the test bundles
+ * 2 - Error writing output file
+ */
+__attribute__((constructor))
+void read_it() {
+    NSString *outputFilePath = [[NSProcessInfo processInfo] environment][@"_BP_TEST_LIST_FILE"] ?: @"/tmp/test_list.json";
+    NSLog(@"[BPTestDumper] Output test dump file path is %@", outputFilePath);
+    NSBundle *mainBundle = [NSBundle mainBundle];
+    NSString *pluginsPath = [mainBundle builtInPlugInsPath];
+    NSArray *xcTests = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:pluginsPath error:Nil];
+    NSLog(@"[BPTestDumper] Got a total of %lu xctest bundles.", [xcTests count]);
+    for (NSString *xcTestBundlePath in xcTests) {
+        NSString *ext = [xcTestBundlePath pathExtension];
+        if (![ext isEqual:@"xctest"]) {
+            NSLog(@"[BPTestDumper] Skipping %@ - %@", xcTestBundlePath, ext);
+            continue;
+        }
+        NSString *path = [pluginsPath stringByAppendingPathComponent:xcTestBundlePath];
+        NSLog(@"[BPTestDumper] Loading %@", path);
+        NSBundle *bundle = [NSBundle bundleWithPath:path];
+        if (!bundle) {
+            printf("[BPTestDumper] bundle failed to load!\n");
+            perror("bundle");
+            exit(1);
+        }
+        [bundle load];
+    }
+    NSLog(@"[BPTestDumper] Creating a default test suite with all suites of suites...");
+    XCTestSuite *defaultTestSuite = [XCTestSuite defaultTestSuite];
+    if (!defaultTestSuite) {
+        printf("[BPTestDumper] Failed to create a defaultTestSuite\n");
+        exit(1);
+    }
+    NSLog(@"[BPTestDumper] Successfully created a defaultTestSuite!!!");
```

Comment:
Three exclamation marks!!! Much excited!!! :-)

Reply:
Yeah, that's a reason to celebrate. Toned it down, though.